specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -18,6 +18,61 @@
     "application/json"
   ],
   "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.RecoveryServices/locations/{location}/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "CheckResourceNameAvailability"
+        ],
+        "summary": "API to check for resource name availability.\r\nA name is available if no other resource exists that has the same SubscriptionId, Resource Name and Type\r\nor if one or more such resources exist, each of these must be GCed and their time of deletion be more than 24 Hours Ago",
+        "operationId": "CheckResourceNameAvailability_Post",

```

Comment:
This will generate a client named `CheckResourceNameAvailability` with one method `Post` which isn't a very good user experience.  A better name would be something like `RecoveryServices_CheckNameAvailability`.

Reply:
Done in New PR: https://github.com/Azure/azure-rest-api-specs/pull/5496